### DISCUSSION TITLE
download library via non graphical way?

### DISCUSSION THREAD

**Discussion by podhorsky-ksj** [immich-app/immich#10712] [Q&A]
Hi,

is there a way how to download library (e.g. in zip file) without gui?

**Comment by bo0tzz** [comment]
You can just copy the files out of UPLOAD_LOCATION

**Comment by podhorsky-ksj** [comment]
In that case they will have completely different filenames. If you download it graphical way, whey will have always original names.
I need something similar to the graphical way.

**Reply by bo0tzz** [reply]
If you want the original filenames then take a look at https://immich.app/docs/administration/storage-template

**Comment by podhorsky-ksj** [comment]
So, reinit immich with storage template? If I will do just the filenames, I will not them paired with the album i want to download. And I'm thinking, how it will behave if I add {{album}} variable? It will move the file to directory with name of album, to which it was moved in gui? And how about files, which are in multiple albums?

**Comment by podhorsky-ksj** [comment]
I tested it and it seems to be like I expected. Once files are uploaded, they never changes directory in which are. So just according to physical location is impossible to determine, which one belongs to which album. These data are in database.

I would like to have download feature  e.g in immich cli. But there is only upload.

**Comment by podhorsky-ksj** [comment]
I have this problem. I like to use immich, but I don't belive it for pernament storage yet.
So, I'm using it for temporary data storage and backup from android etc.

This way, I need now to download a library to external pc first and copy it back to server, which is very slow and I need to be on place with higher network speed or close to server.

I keep server cli only, so I can't download it directly on server. I have configured aria on server, so, I can send links, which will be downloaded on this server directly. But in Immich in gui is only button to download library in zip, which doesn't provide link, which could be send to aria.

Other way would be have browser in server and connect to it via ssh -X, but that is slow and I was glad, when I replaced it by aria.

**Comment by podhorsky-ksj** [comment]
Just checked how is the library downloaded via dev tools, that there is sent POST request with json with all media assets to immich server. There is no way how to do it via aria from my side.
